packages/client-core/src/common/services/ProjectService.ts

```diff
@@ -73,17 +77,73 @@ export const ProjectService = {
   invalidateProjectCache: async (projectName: string) => {
     try {
       await API.instance.client.service('project-invalidate').patch({ projectName })
-      ProjectService.fetchProjects()
+      await ProjectService.fetchProjects()
     } catch (err) {
       logger.error(err, 'Error invalidating project cache.')
     }
+  },
+
+  setRepositoryPath: async (id: string, url: string) => {
+    try {
+      await API.instance.client.service('project').patch(id, {
+        repositoryPath: url
+      })
+    } catch (err) {
+      logger.error(err, 'Error setting project repository path')
+      throw err
+    }
+  },
+
+  pushProject: async (id: string) => {
+    try {
+      await API.instance.client.service('project-github-push').patch(id, {})
+    } catch (err) {
+      logger.error('Error with project push', err)
+      throw err
+    }
+  },
+
+  createPermission: async (userInviteCode: string, projectId: string) => {
+    try {
+      await API.instance.client.service('project-permission').create({
+        inviteCode: userInviteCode,
+        projectId: projectId
+      })
+    } catch (err) {
+      logger.error('Error with creating new project-permission', err)
+      throw err
+    }
+  },
+
+  patchPermission: async (id: string, type: string) => {
+    try {
+      await API.instance.client.service('project-permission').patch(id, {
+        type: type
+      })
+    } catch (err) {
+      logger.error('Error with patching project-permission', err)
+      throw err
+    }
+  },
+
+  removePermission: async (id: string) => {
+    try {
+      await API.instance.client.service('project-permission').remove(id)
+    } catch (err) {
+      logger.error('Error with removing project-permission', err)
+      throw err
+    }
   }
 }
 // TODO
 // client.service('project-build').on('patched', (params) => {
 //   store.dispatch(ProjectAction.buildProgress(params.message))
 // })
 
+API.instance.client.service('project').on('patched', (params) => {
```

Comment:
this needs to be moved to the service in a `useAPIListeners()` as elsewhere in the codebase

Reply:
Check now, I think I did it right.

Reply:
almost, need to pull the listener out and clean it up on the useEffect unmount return function